pkg/gatewayserver/io/ws/lbslns/upstream.go

```diff
@@ -583,8 +583,18 @@ func (f *lbsLNS) HandleUp(ctx context.Context, raw []byte, ids *ttnpb.GatewayIde
 		updateSessionID(ctx, SessionIDFromXTime(jreq.UpInfo.XTime))
 		ct := recordTime(jreq.RefTime, jreq.UpInfo.XTime, jreq.UpInfo.GPSTime, jreq.UpInfo.RxTime)
 		if err := conn.HandleUp(up, ct); err != nil {
-			logger.WithError(err).Warn("Failed to handle upstream message")
-			return nil, err
+			// HandleUp calls uplink.ValidateFields which is a deep validation of the payload fields.
+			// jreq.toUplinkMessage already validates that the gateway is sending sane messages.
+			// Ex: The MHDR of a `jreq` message is valid LoRaWAN Join type.
+			// But beyond these basic checks, LBS gateways do not perform a deep validation of the message.
+			// Ex: Checking that the RFU fields are not used is out of scope for the gateway and the Gateway Server.
+			// In such cases, we won't process the uplink message but we shouldn't disconnect the gateway.
+			// This prevents a spamming device from DOSing the gateway.
+			if !errors.IsInvalidArgument(err) {
```

Comment:
This is one of the cases when I think that error propagation shouldn't be done anyway.

Virtually the only error that we may encounter is a validation error, or a buffer full error, so just leaving the warning in place (in both cases) is fine.